Source/WebCore/Modules/websockets/WebSocketExtensionParser.cpp

```diff
@@ -35,18 +35,16 @@
 #include <wtf/ASCIICType.h>
 #include <wtf/text/CString.h>
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-
 namespace WebCore {
 
 bool WebSocketExtensionParser::finished()
 {
-    return m_current >= m_end;
+    return m_data.empty();
 }
 
 bool WebSocketExtensionParser::parsedSuccessfully()
 {
-    return m_current == m_end;
+    return m_data.empty() && !m_didFailParsing;
```

Comment:
I wonder why we need an explicit m_didFailParsing bit, instead of just checking for empty() like the code used to. I don't think it's bad, but ideally a bounds safety transformation wouldn't need it.

Reply:
It used to be `m_current == m_end` is finished successfully and `m_current == m_end + 1` failed, based on how `finished()` and `parsedSuccessfully()` differ. I could not represent current going past the end with a span.